crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -634,47 +634,42 @@
 //! [ruma-types]: https://docs.rs/ruma/latest/ruma/api/client/sync/sync_events/v4/index.html
 //! [future-signals-tutorial]: https://docs.rs/futures-signals/latest/futures_signals/tutorial/index.html
 
+mod client;
+mod config;
+mod room;
+mod view;
+
 use std::{
     collections::BTreeMap,
     fmt::Debug,
-    ops::{Deref, Not},
+    ops::Deref,
     sync::{
-        atomic::{AtomicBool, AtomicU8, Ordering},
+        atomic::{AtomicU8, Ordering},
         Arc, Mutex,
     },
     time::Duration,
 };
 
+pub use client::*;
+pub use config::*;
 use futures_core::stream::Stream;
-use futures_signals::{
-    signal::Mutable,
-    signal_map::{MutableBTreeMap, MutableBTreeMapLockRef},
-    signal_vec::{MutableVec, MutableVecLockMut},
-};
-use matrix_sdk_base::{deserialized_responses::SyncTimelineEvent, sync::SyncResponse};
+use futures_signals::{signal::Mutable, signal_map::MutableBTreeMap, signal_vec::MutableVec};
+pub use room::*;
```

Comment:
I'm not sure about the wildcard re-exports. Maybe expand them?

Reply:
If it's public, it's gonna be re-exported, otherwise it will stay private (or public for the particular specified path).

Reply:
I just mean using literal `*` instead of naming the individual items. I find the expanded re-exports much clearer when reading (you get a sense of what is being re-exported). There's a clippy lint for it but we don't activate it. https://rust-lang.github.io/rust-clippy/master/index.html#wildcard_imports

I think the only other places we do this is ffi crates and test modules (and I would like to have expanded imports there too).